XcodeAutoBasher/XcodeAutoBasher/Models/VOKScriptForFolder.m

```diff
@@ -155,10 +155,13 @@ - (void)actuallyRunScript
 {
     //Check if file is runnable or Xcode will crash
     if ([[NSFileManager defaultManager] isExecutableFileAtPath:self.absolutePathToScript]) {
-        NSTask *task = [[NSTask alloc] init];
-        task.launchPath = self.absolutePathToScript;
-        task.currentDirectoryPath = [task.launchPath stringByDeletingLastPathComponent];
-        [task launch];
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+            NSTask *task = [[NSTask alloc] init];
+            task.launchPath = self.absolutePathToScript;
+            task.currentDirectoryPath = [task.launchPath stringByDeletingLastPathComponent];
+            task.environment = self.containingProject.environmentVariables;
```

Comment:
First, keep in mind that the main thread here is Xcode's UI thread.  Really, nothing that isn't touching the UI should be run on that thread.

The task was never being run on the main thread, since it's being launched asynchronously with `NSTask`, but it's also an independent process in a separate environment, so its execution is decoupled from Xcode's threads.  From the perspective of whatever's being run in the `NSTask`, there's no difference between creating and launching it on Xcode's UI thread or on a background thread, especially since we're not waiting for it to finish or doing anything to the UI as a result of its execution.